test/infoblox_test.rb

```diff
@@ -20,7 +20,7 @@ def test_create_a
     ip = '10.1.1.1'
 
     @provider.expects(:a_record_conflicts).with(fqdn, ip).returns(-1)
```

Comment:
Here and in the rest of the tests below: the super class takes care of checks for conflicts for us and has tests to verify these checks are called (https://github.com/theforeman/smart-proxy/blob/develop/test/dns_common/record_test.rb#L174). All infoblox provider tests need to verify is that when do_create is called, an ib_create_*/ib_delete_* appropriate for the record type is called. These should limit each test to a single expectation.